### PR TITLE
Use Makefile to centralize build/test commands

### PR DESCRIPTION
**Pull request by andyleejordan**
### Explicitly update developer tool packages
Biggest change is the move to Black 20.8b1. W503 is now disabled by
default, and since we control the versions of the tools we use, we can
safely update our configuration.
### Enable cyclomatic complexity checking
And disable existing violations with noted TODOs.
### Update to Poetry version 1.1.0b4

### CODE REVIEW

File: lisa/testselector.py [hunk 1]
```diff
@@ -139,13 +139,14 @@ def _force_check(
     return is_skip
 
 
-def _apply_filter(
+def _apply_filter(  # noqa: C901
```
Comment (squirrelsc) [inline lisa/testselector.py:142]: What's this?

**Review by squirrelsc — APPROVED** [review]

**Review by lpereira — CHANGES_REQUESTED** [review]
There are lots of unrelated changes in this commit due to a change in the flake8 config, I'd rather see them in another PR.

**Comment by squirrelsc** [comment]
> There are lots of unrelated changes in this commit due to a change in the flake8 config, I'd rather see them in another PR.

This PR has meaningful commits, that may be easier for reviewing. If it's merged by rebase, it would help on understanding it in future.

**Comment by andyleejordan** [comment]
@lpereira the changes are broken down per-commit, much better to view than GitHub's default of showing everything together.

**Review by lpereira — APPROVED** [review]

File: lisa/testselector.py (hunk 1 above)
Comment (andyleejordan) [reply lisa/testselector.py:142]: It's the flake8 way of disabling a warning for a particular line of code. In this case, C901 is the complexity warning that was enabled. I'd like the setting turned on so we can be see the warning on new code, but don't feel that we need to correct the existing violations. (Also, "noqa" probably comes from "no quality assurance.")

**Comment by andyleejordan** [comment]
Thanks!

**Pull request merged**